### PR TITLE
923 vds centreline equivalency table

### PR DESCRIPTION
**Pull request by gabrielwol**
## What this pull request accomplishes:

What started as adding centreline table snowballed to updates of vds.detector_inventory and a whole host of smaller updates. I've added checkboxes for the reviewer to keep track!

- [ ] Adds new `vds.centreline_vds` table:
  - I also manually assigned a bunch of centreline_ids using QGIS as part of this PR, so we now have centreline_ids for every sensor for which we have data (division_id 2 only). 
  - included readme updates and a script + img to help identify new sensors' centreline_ids: `sql/adhoc_updates/update-missing_vds_centreline.sql` (not automated)
  
- [x] adds a weekly `vds_check` DAG to notify if there are missing centreline_ids or `expected_bins` values for new sensors
  - `volumes/vds/sql/select/select-missing_centreline.sql`, `volumes/vds/sql/select/select-missing_expected_bins.sql`
  
- [x] Big updates view `vds.detector_inventory`:
  - Base off of new `vdsconfig_x_entity_locations` which is all combinations of vdsconfig_uid and entity_location_uid that appear in the data. This new table enabled next point:
  - added first and last active date (gets updated every day via `vdsconfig_x_entity_locations`
  - added description
  - added geom, centreline_id and geom
  - I think this should make DRDs lives a lot easier!!! 

- [ ] these 4 sqls were moved into their own sqls to clean up python/vds_functions.py: 
  - volumes/vds/sql/insert/insert_commdeviceconfig.sql
  - volumes/vds/sql/insert/insert_entity_locations.sql - **Note:** also added update geom to this script
  - volumes/vds/sql/insert/insert_raw_vdsdata.sql - **Note:** to this one I also added an upsert to new `vdsconfig_x_entity_locations` based on inserted values.
  - volumes/vds/sql/insert/insert_vdsconfig.sql

## Issue(s) this solves:

- #923 

## What, in particular, needs to reviewed:

- the naming and method of `vdsconfig_x_entity_locations`. I went with a table updated daily to stop a very large mat view from having to be refreshed daily. 
- `vds.detector_inventory`: I've thought about making this a table instead of a view, but we would loose all the info contained in the case statements? the `expected_bins` column is referenced by a number of other queries, which is also why I left the duplicate uid column, to not break those joins. 

## What needs to be done by a sysadmin after this PR is merged
- files need to be checked out in morbius data_scripts
  - update link to 2 DAGs

### CODE REVIEW

**Comment by chmnata** [comment]
image
Checked all point's colour generally matched with the line's colour. Spot checked some that looks off but they are all good!

**Review by chmnata — COMMENTED** [review]
Lightly reviewed, centreline + dags looks good!

File: dags/vds_pull_vdsdata.py [hunk 1]
```diff
@@ -30,7 +30,7 @@
     'start_date': datetime(2021, 11, 1),
     'email_on_failure': False,
     'email_on_retry': False,
-    'retries': 5,
+    'retries': 1,
```
[the remark this answers is not in the document]
Comment (gabrielwol) [reply dags/vds_pull_vdsdata.py:33]: nah, me realizing 5 retries is crazy

File: dags/vds_check.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        external_task_id="done",
+        poke_interval=3600, #retry hourly
+        mode="reschedule",
+        timeout=86400, #one day
+        execution_delta=timedelta(days=-6, hours=1) #pull_vds scheduled at '0 4 * * *'
+    )
+
+    check_missing_centreline_id = SQLCheckOperatorWithReturnValue(
+        task_id="check_missing_centreline_id",
+        sql="select-missing_centreline.sql",
```
[the remark this answers is not in the document]
Comment (gabrielwol) [reply dags/vds_check.py:68]: Good idea, can implement if there are a lot of new sensors.

**Review by chmnata — APPROVED** [review]

**Pull request merged**